monkey/monkey_island/cc/services/log_service/flask_resources/island_log.py

```diff
@@ -21,5 +22,8 @@ def __init__(self, island_log_file_path: Path):
     def get(self):
         try:
             return get_text_file_contents(self._island_log_file_path)
+        except FileNotFoundError:
+            logger.error("Monkey Island logs do not exist", exc_info=True)
+            return "", HTTPStatus.NOT_FOUND
```

Comment:
I think this check is erroneous. If the Island is running, it should have a log. if we can't return that log for any reason, that's some kind of internal server error. The solution is probably to reraise in the `except Exception` clause.